blocks/banner/banner.js

```diff
@@ -0,0 +1,61 @@
+import {
+  createOptimizedPicture,
+  normalizeHeadings,
+} from '../../scripts/utils.js';
+
+export default async function init(block) {
+  const bannerContents = document.createElement('div');
+  bannerContents.classList.add('banner-contents');
+  block.querySelectorAll('a').forEach(async (a) => {
+    if (!a?.href) return;
+    // content wrapper
+    const bannerContent = document.createElement('div');
+    bannerContent.classList.add('content-wrapper');
+
+    try {
+      // get response from the URL
+      const { pathname } = new URL(a);
+      const path = pathname?.replace(/\.html$/, '');
+      if (path) {
+        const response = await fetch(`${path}.plain.html`);
+        if (response.ok) {
+          const responseEl = document.createElement('div');
+          responseEl.innerHTML = await response.text();
+          block.classList.add('is-loaded');
+
+          // creating banner image and text div.
+          const bannerImage = document.createElement('div');
+          const bannerText = document.createElement('div');
+          bannerImage.classList.add('banner-image');
+          bannerText.classList.add('banner-text');
+
+          // banner image content
+          const img = responseEl.querySelector('img');
+          const picture = img.closest('picture');
+          const newPicture = createOptimizedPicture(img.src, img.alt);
+          if (picture) {
+            picture.parentElement.replaceChild(newPicture, picture);
+            bannerImage.append(newPicture);
+          }
+
+          // banner text content
+          normalizeHeadings(responseEl, ['h3']);
+          const link = responseEl.querySelector('a');
+          link.classList.add('cta-link');
+          bannerText.append(responseEl);
+
+          // appending DOM objects
+          bannerContent.append(bannerImage);
+          bannerContent.append(bannerText);
```

Comment:
Element.append() can append several nodes and strings.

```suggestion
          bannerContent.append(bannerImage, bannerText);
```

Reply:
nice catch! Addressed in https://github.com/adobecom/blog/pull/13/commits/45863e761cfeaaa139bac5e18c4191d9778d82ff